### ISSUE TITLE
Apply dep-profiles with error: define dep profile: Put https://domain/v1/dep/profiles:stream error: stream ID 1; INTERNAL_ERROR

### ISSUE THREAD

**Issue by windliyue** [micromdm/micromdm#663]
### What version of `micromdm` are you using?
v1.6.0
### What `micromdm` command did you run?
mdmctl apply dep-profiles -f /path/to/dep-profile.json

**Comment by jessepeterson** [comment]
This seems like an HTTPS error. Are you still getting this problem? Are you using MicroMDM directly or do you have a reverse proxy in front of it? What does your config in `~/.micromdm/servers.json` look like?

**Comment by williamtheaker** [comment]
I'm seeing this myself when interacting directly with an instance

server logs:
```
Apr 26 23:17:09 ip-10-0-49-27 micromdm[270515]: transport=http method=PUT status=200 proto=HTTP/2.0 host=REDACTED_IP_ADDRESS user_agent=Go-http-client/2.0 path=/v1/dep/profiles
Apr 26 23:17:09 ip-10-0-49-27 micromdm[270515]: ts="2021/04/26 23:17:09" caller="http2: panic serving REDACTED_IP_ADDRESS" msg="runtime error: invalid memory address or nil pointer dereference
```

servers.json
```
{
  "active": "xx",
  "servers": {
    "xx": {
      "api_token": "redacted",
      "server_url": "https://xx.redacted-domain.tld/",
      "skip_verify": false
    }
  }
}
```

An unmodified DEP profile produced by `mdmctl apply dep-profiles -template` gives the same error.

**Comment by jessepeterson** [comment]
@williamtheaker you  get this exact error? i can't duplicate this - i generate a new DEP profile with `-template`, don't modify it  at  all, and sent it back up with `mdmctl apply dep-profiles -f <just-generated-with-template-switch.json>` and it works fine. Here's the DEP  profile `-template` generated:

```
% ./build/darwin/mdmctl apply dep-profiles -template
{
  "profile_name": "(Required) Human readable name",
  "url": "https://mymdm.example.org/mdm/enroll",
  "allow_pairing": true,
  "is_supervised": false,
  "is_multi_user": false,
  "is_mandatory": false,
  "await_device_configured": false,
  "is_mdm_removable": true,
  "support_phone_number": "(Optional) +1 408 555 1010",
  "support_email_address": "(Optional) support@example.com",
  "org_magic": "(Optional)",
  "anchor_certs": [],
  "supervising_host_certs": [],
  "skip_setup_items": ["AppleID", "Android"],
  "department": "(Optional) support@example.com",
  "devices": ["SERIAL1","SERIAL2"]
}
```

And when save that (exactly) and apply  it:

```
 % ./build/darwin/mdmctl apply dep-profiles -f tpl.json
Defined DEP Profile with UUID A476A909F05056EB3765AC2A3BC4B36F
```

**Comment by williamtheaker** [comment]
Yep when I run: `mdmctl apply dep-profiles -f ~/Downloads/build/darwin/dep-profile.json -filter='*'`
I get: `define dep profile: Put "https://xx.redacted-domain.tld/v1/dep/profiles": stream error: stream ID 1; INTERNAL_ERROR`

I am seeing this on a few instances, while others aren't impacted. I need to isolate a specific version number and see if I can replicate it again. I *think* this is only happening on instances where I've already generated/downloaded the DEP public certificate without first trying to apply a DEP profile. i.e. this has only occurred on my "real production" instances, but I need to validate that.

**Comment by williamtheaker** [comment]
Well I figured out the issue: you can't apply a DEP profile before uploading the .p7m server token file from ABM and completing the DEP trust relationship.

Is there a better way to expose this error either with mdmctl or micromdm's logs?

**Comment by jessepeterson** [comment]
I still can't repro this exactly as you're seeing it. But I found a bug while exploring it (see the linked issue). Also I was reminded that while one has `-http-debug` on `mdmctl` error reporting  is broken (because HTTP statuses get changed for some reason) which will hide API errors — might you have HTTP debugging switched on?

**Comment by williamtheaker** [comment]
I get the same output no matter if `-http-debug=` `true`,`false`, or omitted entirely:

```
% mdmctl apply dep-profiles -f example.json -filter='*'
define dep profile: Put "https://fqdn/v1/dep/profiles": stream error: stream ID 1; INTERNAL_ERROR
% mdmctl apply dep-profiles -f example.json -filter='*'
define dep profile: Put "https://fqdn/v1/dep/profiles": stream error: stream ID 1; INTERNAL_ERROR
% mdmctl apply dep-profiles -f example.json -filter='*'
define dep profile: Put "https://fqdn/v1/dep/profiles": stream error: stream ID 1; INTERNAL_ERROR
```

**Comment by jessepeterson** [comment]
@williamtheaker would you be able to test if #750 addresses this for you?

**Comment by williamtheaker** [comment]
Built from your branch and still get the error:
`transport=http method=PUT status=200 proto=HTTP/2.0 host= user_agent=Go-http-client/2.0 path=/v1/dep/profiles
ts="2021/05/03 23:38:30" caller="http2: panic serving " msg="runtime error: invalid memory address or nil pointer dereference"`

To be clear: this isn't a real issue applying DEP profiles. Once you've uploaded the DEP token to micromdm, this works as expected.

**Comment by jessepeterson** [comment]
Hmm.. I can't duplicate this now. Do you have a set of reproduction steps? What I did: removed my oauth tokens from the boltDB to get a similar panic (which 750 fixed).

**Comment by jessepeterson** [comment]
@williamtheaker can you give this another go off the main branch now that #750 has been merged? I want to get to the bottom of this but I just can't repro now that that has been fixed.
